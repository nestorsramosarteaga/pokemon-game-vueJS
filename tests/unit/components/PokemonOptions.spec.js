import { shallowMount } from "@vue/test-utils"
import PokemonOptions from "@/components/PokemonOptions"

import { pokemons } from "../mocks/pokemons.mock"

describe("PokemonOptions Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons,
      },
    });
  });

  test("Debe de hacer match con el snapshot", () => {
    
    //console.log(wrapper.html());

    // toMatchInlineSnapshot
    expect( wrapper.html() ).toMatchSnapshot()
    
  });

  test('Debe de mostrar las 4 opciones correctamente', () => {
    const liOptions = wrapper.findAll("li");
    expect(liOptions.length).toBe(4);

    const list = liOptions.map( (wrp) => wrp.text() )
    //console.log(list)   

    const list2 = pokemons.map( ({name}) => name)
    //console.log(list2)

    expect(list).toEqual(list2);

    expect( liOptions[0].text() ).toBe('bulbasaur')
    expect( liOptions[1].text() ).toBe('charmander')
    //...
  })


  test('Debe emitir "selection" con sus respectivos parÃ¡metors al hacer click', () => {
    const [li1, li2, li3, li4] = wrapper.findAll("li");

    li1.trigger('click')
    li2.trigger('click')
    li3.trigger('click')
    li4.trigger('click')

    //console.log( wrapper.emitted('pokemonSelection') )
    expect( wrapper.emitted('pokemonSelection').length ).toBe(4)
    expect( wrapper.emitted('pokemonSelection')[0] ).toEqual([1])
    expect( wrapper.emitted('pokemonSelection')[1] ).toEqual([4])
    expect( wrapper.emitted('pokemonSelection')[2] ).toEqual([3])
    expect( wrapper.emitted('pokemonSelection')[3] ).toEqual([2])

  })


});
