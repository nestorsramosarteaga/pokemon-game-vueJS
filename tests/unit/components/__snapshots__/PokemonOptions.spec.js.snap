// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component Debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>bulbasaur</li>
    <li>charmander</li>
    <li>venusaur</li>
    <li>ivysaur</li>
  </ul>
</div>
`;
